src/tasks/dump_db/dump-export.sql.hbs

```diff
@@ -1,19 +1,7 @@
-BEGIN;
+BEGIN ISOLATION LEVEL REPEATABLE READ, READ ONLY;
```

Comment:
This provides somewhat weaker consistency guarantees for the snapshot used for the database dump. The Postgres documentation explicitly recommends using `SERIALIZABLE READ ONLY DEFERRABLE` for backups.

We don't seem to have a very strong case for moving the dump to the read-only replica, since it doesn't seem to have cause any problems on the main replica. Neither does it seem unreasonable to do so, even with the weaker consistency guarantees.

Reply:
One more question – are we sure `REPEATABLE READ` will work on the hot spare? It isn't obvious why `SERIALIZABLE READ ONLY` would not work, so we should test this on stage.

Reply:
> are we sure `REPEATABLE READ` will work on the hot spare?

I definitely expect this to work.  We can test by deploying the code, manually scheduling an export, verifying that everything works, and then finally updating the daily task.

> It isn't obvious why `SERIALIZABLE READ ONLY` would not work

This definitely was a surprise to me as well.  But I hope my other response helps explain why it isn't possible for the read-replica to be directly involved in cycle tracking on the leader.  I've seen several suggestions that using `DEFERABLE` from a replica could be supported in future releases by writing some additional data in the WAL so that the follower can find a valid snapshot to start from.  It appears no such enhancement has been implemented yet.

> we should test this on stage.

We can't actually test this on staging, because Heroku only supports read-only replicas on paid database plans.  But we shouldn't have any issues merging and deploying this change as we can roll it out via a configuration change to the scheduler.

> We don't seem to have a very strong case for moving the dump to the read-only replica, since it doesn't seem to have cause any problems on the main replica.

I'm not entirely sure about that.  I don't think any of the temporary response time spikes can be traced to the the export specifically, but I've definitely seen slight performance improvements and much less variation in response times after moving just a few expensive endpoints (#2073) to the read-only replica.  In total, that PR seems to have moved only around 600 queries per hour off of the primary database, but the improvements seem impressive.  It also appears to have eliminated the occasional spikes in download response times.  I've collected some rough numbers and plan to follow up on that PR with more details, but I would definitely prefer to run the export from there as well.

Reply:
I'm perfectly fine with moving to the read replica, and with the additional explanations you gave I'm actually actively in favour of doing so.